arkouda/categorical.py

```diff
@@ -159,7 +159,32 @@ def __str__(self):
     def __repr__(self):
         return "array({})".format(self.__str__())
 
-    def binop(self, other, op):
+    def binop(self, other : 'Categorical', op : str) -> pdarray:
```

Comment:
`binop` is not suppose to be user facing. maybe it should get underscores prepended to the name?

Reply:
Good idea, I agree

Reply:
fixed!